src/main/java/org/mskcc/cmo/metadb/service/impl/ValidRequestCheckerImpl.java

```diff
@@ -0,0 +1,268 @@
+package org.mskcc.cmo.metadb.service.impl;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.lang3.EnumUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.logging.log4j.util.Strings;
+import org.mskcc.cmo.common.enums.CmoSampleClass;
+import org.mskcc.cmo.common.enums.SampleOrigin;
+import org.mskcc.cmo.common.enums.SampleType;
+import org.mskcc.cmo.common.enums.SpecimenType;
+import org.mskcc.cmo.metadb.service.ValidRequestChecker;
+import org.mskcc.cmo.metadb.service.util.RequestStatusLogger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ValidRequestCheckerImpl implements ValidRequestChecker {
+
+    private final ObjectMapper mapper = new ObjectMapper();
+    private boolean isCmoRequest;
```

Comment:
do these need to be global?

if you need them in other methods then you can pass them as an argument to the method

Reply:
since this request checker class will be validating multiple request jsons then I would feel safer about having these only as local variables in the main method called in this class